iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/HelmEvaluator.java

```diff
@@ -99,13 +99,16 @@ void writeTemplateAndDependencies(Process process, String name, String content,
       for (var filenameToFileContent : templateDependencies.entrySet()) {
         writeFileToProcess(out, filenameToFileContent.getKey(), filenameToFileContent.getValue());
       }
-      writeStringAsBytes(out, String.format("END%n"));
+      writeStringAsBytes(out, "END\n");
     }
   }
 
+  // Suppress java:S3457 - Format strings should be used correctly.
+  // Here \n (instead of %n) needs to be used as it is expected on Go site
+  @SuppressWarnings("java:S3457")
   private static void writeFileToProcess(OutputStream out, String fileName, String content) throws IOException {
-    writeStringAsBytes(out, "%s%n".formatted(fileName));
-    writeStringAsBytes(out, "%d%n".formatted(content.lines().count()));
+    writeStringAsBytes(out, "%s\n".formatted(fileName));
+    writeStringAsBytes(out, "%d\n".formatted(content.lines().count()));
     if (!content.endsWith("\n")) {
       content += "\n";
     }
```

Comment:
Do we need to do also anything about `writeStringAsBytes(out, content)`? E.g. make sure all lines end with `\n`? Can it make things easier on the Go side?

Reply:
I considered replacing all line separators by `\n`, but it was a dirty solution for me because the evaluated output will be not the same as produced by `helm` command. It will also increase the location shifting mechanism complexity.

Reply:
Ok, probaly I caught the idea: https://github.com/SonarSource/sonar-iac/pull/1215#discussion_r1542557046

Reply:
But please add comments in HelmEvaluator and in io-handling about why we only "normalize" file name and lines count and not the content.